sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -147,22 +148,35 @@ AZ_NODISCARD AZ_INLINE az_result az_iot_pnp_client_get_sas_signature(
  * @param[in] base64_hmac_sha256_signature The Base64 encoded value of the HMAC-SHA256(signature,
  *                                         SharedAccessKey). The signature is obtained by using
  *                                         #az_iot_pnp_client_sas_signature_get.
+ * @param[in] token_expiration_epoch_time The time, in seconds, from 1/1/1970.
  * @param[in] key_name The Shared Access Key Name (Policy Name). This is optional. For security
  *                     reasons we recommend using one key per device instead of using a global
  *                     policy key.
- * @param[in] mqtt_password An empty #az_span with sufficient capacity to hold the MQTT password.
- * @param[out] out_mqtt_password The output #az_span containing the MQTT password.
+ * @param[out] mqtt_password A buffer with sufficient capacity to hold the MQTT password.
+ *                           If successful, contains a null-terminated string with the password
+ *                           that needs to be passed to the MQTT client.
+ * @param[in] mqtt_password_size The size, in bytes of \p mqtt_password.
```

Comment:
The docs for `mqtt_password_size` and `out_mqtt_password_length` are confusing and it is hard to distinguish between the two.

Consider being explicit that `mqtt_password_size` is the maximum available space that in the buffer that this method can write to.

For example, see https://github.com/Azure/azure-sdk-for-c/pull/590#discussion_r414204116

https://github.com/Azure/azure-sdk-for-c/blob/22b3f211593beac296257f12603094fb0b6979db/sdk/core/core/inc/az_span.h#L217-L221